### PR TITLE
optimize searchd index

### PR DESCRIPTION
**Pull request by akostadinov**
Adds a task to optimize searchd index and also allow tuning
the optimization thread performance.

These variables will only help during container build when we generate
the sphinx config file. Searchd container will pick up the file and
will ignore any such variables.

Also removed openshift `thinking_sphinx.yml` override as I didn't see any differences with the default one.

### CODE REVIEW

File: config/thinking_sphinx.yml [hunk 1]
```diff
@@ -62,4 +70,5 @@ test: &test
 
 production:
   <<: *default
+  skip_running_check: true # daemon is remote in production
```
Comment (akostadinov) [inline config/thinking_sphinx.yml:73]: This is to allow us running `ts:index` without funny workarounds. I spotted this just now :face_with_head_bandage:

**Comment by mayorova** [comment]
@akostadinov 

One of the differences in OpenShift variant of `thinking_sphinx.yml` is that it has:
```
  # Use standard output for logs in container environments
  query_log: /dev/stdout
  log: /dev/stdout
```

Don't know, maybe at least we need to put it under `production` env section, if you really don't want to keep the two files.

**Comment by akostadinov** [comment]
my bad, let me have a look, how did I miss this

File: lib/tasks/searchd.rake [hunk 2]
```diff
@@ -34,4 +34,26 @@ namespace :sphinx do
...
+      # no way to wait for an index optimization to finish before starting another unless we switch to manticore
+      ThinkingSphinx::Connection.take do |connection|
+        connection.execute "OPTIMIZE INDEX #{index.name}"
+      end
+
+      warn "#{index.name} enqueued for optimization"
+    end
+  end
+
+  desc "index and optimize remote search indices"
```
Comment (mayorova) [inline lib/tasks/searchd.rake:57]: Why "remote" in the description? it will work for both local and remote.

**Comment by akostadinov** [comment]
@mayorova , addressed your comments, thanks a lot!

**Review by mayorova — APPROVED** [review]
Approving, but I'm not convinced by not having default values for `log` and `query_log` in production.

I guess I will restore customization on 3scale-saas until we fully migrate to searchd.

**Pull request merged**